tools/generator/test/XcodeScheme+ExtensionsTests.swift

```diff
@@ -0,0 +1,371 @@
+import XCTest
+
+@testable import generator
+
+// TODO: Add some testActions
+
+// MARK: allSchemeLabels Tests
+
+extension XcodeSchemeExtensionsTests {
+    func test_allSchemeLabels_doNotOverwriteTopLevel() throws {
+        // Ensure that toolLabel comes through as top-level even though it
+        // is specified in build action as well.
+        let scheme = XcodeScheme(
+            name: "Foo",
+            buildAction: .init(targets: [libLabel, toolLabel]),
+            testAction: nil,
+            launchAction: .init(target: toolLabel, args: [], env: [:], workingDirectory: nil)
+        )
+        let actual = scheme.allSchemeLabels
+        let expected: Set<XcodeScheme.SchemeLabel> = [
+            .init(label: libLabel, isTopLevel: false),
+            .init(label: toolLabel, isTopLevel: true),
+        ]
+        XCTAssertEqual(expected, actual)
+    }
+
+    func test_allSchemeLabels_forToolScheme() throws {
+        let actual = toolScheme.allSchemeLabels
+        let expected: Set<XcodeScheme.SchemeLabel> = [
+            .init(label: libLabel, isTopLevel: false),
+            .init(label: toolLabel, isTopLevel: true),
+        ]
+        XCTAssertEqual(expected, actual)
+    }
+}
+
+// MARK: resolveTargetIDs Tests
+
+extension XcodeSchemeExtensionsTests {
+    func test_resolveTargetIDs_withToolScheme() throws {
+        let actual = try toolScheme.resolveTargetIDs(targets: targets)
+        let expected = [
+            libLabel: libmacOSx8664TargetID,
+            toolLabel: toolmacOSx8664TargetID,
+        ]
+        XCTAssertEqual(expected, actual)
+    }
+
+    func test_resolveTargetIDs_withIOSAppScheme() throws {
+        // Both the device and simulator TargetID values are available.
+        // Prefer the TargetID values for the simulator.
+        let actual = try iOSAppScheme.resolveTargetIDs(targets: targets)
+        let expected = [
+            libLabel: libiOSx8664TargetID,
+            iOSAppLabel: iOSAppiOSx8664TargetID,
+        ]
+        XCTAssertEqual(expected, actual)
+    }
```

Comment:
@brentleyjones I just wanted to confirm that we do want to prefer the simulator platform, if both the simulator and device targets are available. The logic `Platform` seems to suggest that is the case. Is that your understanding?

Reply:
Yes